arch/arm/core/Kconfig

```diff
@@ -34,13 +34,28 @@ config CPU_HAS_SYSTICK
 	help
 	  This option is enabled when the CPU has systick timer implemented.
 
+config BUILTIN_STACK_GUARD
+	bool "Thread Stack Guards based on built-in ARM stack limit checking"
+	depends on CPU_CORTEX_M_HAS_SPLIM
+	default n
+	help
+	  Enable Thread/Interrupt Stack Guards via built-in Stack Pointer
+	  limit checking. The functionality must be supported by HW.
+
 config ARM_STACK_PROTECTION
 	bool
 	default y if HW_STACK_PROTECTION
-	select MPU_STACK_GUARD
+	select BUILTIN_STACK_GUARD if CPU_CORTEX_M_HAS_SPLIM
+	select MPU_STACK_GUARD if (!BUILTIN_STACK_GUARD && ARM_CORE_MPU)
```

Comment:
ARCH_HAS_STACK_PROTECTION (which enables HW_STACK_PROTECTION) might, now, be activated, in the wake of _HAS_SPLIM support, regardless of _HAS_MPU support. Therefore, I add the ARM_CORE_MPU in the MPU_STACK_GUARD "select" dependencies for extra protection.

(FYI: config MPU_STACK_GUARD, anyway, depends on ARM_CORE_MPU, we would have a warning anyways...)